src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -0,0 +1,235 @@
+          Failed_Tests++;
+
+        }
+        Total_number_of_tests++;
+      }
+    }
+
+
+
+    /*//get all the test cases in infrastucture module
+    File testFolderInfrastucture = new File("../../../infrastucture/tests");
+    File[] listOfTests1 = testFolderInfrastucture.listFiles();
+    System.out.println("Running all the test cases of infrastucture module...");
+    for(File file : listOfTests1){
+      if(file.isFile()){
+        runByName(file.getName());
+      }
+    }
+    
+    //get all the test cases in networking module
+    File testFolderNetworking = new File("../../../networking/tests");
+    File[] listOfTests2 = testFolderNetworking.listFiles();
+    System.out.println("Running all the test cases of networking module...");
+    for(File file : listOfTests2){
+      if (file.isFile()){
+        runByName(file.getName());
+      }
+    }
+
+    //get all the test cases in processing module
+    File testFolderProcessing = new File("../../../processing/tests");
+    File[] listOfTests3 = testFolderProcessing.listFiles();
+    System.out.println("Running all the test cases of processing module...");
+    for(File file : listOfTests3){
+      if(file.isFile()){
+        runByName(file.getName());
+      }
+    }
+    
+    //get all the test cases in UI module
+    File testFolderUI = new File("../../../UI/tests");
+    File[] listOfTests4 = testFolderUI.listFiles();
+    System.out.println("Running all the test cases of UI module...");
+    for(File file : listOfTests4){
+      if(file.isFile()) {
+        runByName(file.getName());
+      }
+    }
+    */
+}
+
+public String[][] getAllTests(){
+  String InfrastructurePath = "../../../infrastructure/tests";
+    File Infrastucture = new File(InfrastructurePath);
+    String[] Tests1 = Infrastucture.list();
+
+    String ContentPath = "../../../content/tests";
+    File Content = new File(ContentPath);
+    String[] Tests2 = Content.list();
+
+    String NetworkPath = "../../../network/tests";
+    File Network = new File(NetworkPath);
+    String[] Tests3 = Network.list();
+
+    String UIPath = "../../../UI/tests";
+    File UI = new File(UIPath);
+    String[] Tests4 = UI.list();
+
+    return new String [][] {Tests1, Tests2, Tests3, Tests4};
+
+}
+ 
+  
+ /**
+  * Boolean method to give result of the Test Case Class
+  * @param testName    The absoulete path to file name of test case class including .java extension as a string
+  * @return    <code>true</code> if the test case pass 
+  *     and actual output is same as expected output; 
+  *     <code>false</code> otherwise.
+  */
+  public boolean runByName(String testName){
+    String[] arrOfStr = testName.split(".", 2); 
+    String testClassName = arrOfStr[0];
+    Class<?> clazz = Class.forName("testClassName");
+    Object test = clazz.getDeclaredConstructor().newInstance();
+    boolean result = test.run();
+    
+    //after saving the result using logger
+    return result;
```

Comment:
Still need to figure out what happens when they call just this function to run one test. At that time there won't be any need to return the result